projects/ubuntu-24.04/Dockerfile

```diff
@@ -13,15 +13,40 @@ RUN apt-get update && \
 RUN apt-get update && \
     apt-get install -y \
         apt-transport-https \
+        ca-certificates \
         gnupg \
         wget \
         lsb-release \
     && \
     apt-get clean && \
     rm -rf /var/lib/apt/lists/* /tmp/*
 
-RUN wget -qO - https://packages.irods.org/irods-signing-key.asc | apt-key add - && \
-    echo "deb [arch=amd64] https://packages.irods.org/apt/ noble main" | tee /etc/apt/sources.list.d/renci-irods.list
+RUN wget -qO - https://packages.irods.org/irods-signing-key.asc | \
+        gpg \
+            --no-options \
+            --no-default-keyring \
+            --no-auto-check-trustdb \
+            --homedir /dev/null \
+            --no-keyring \
+            --import-options import-export \
+            --output /etc/apt/keyrings/renci-irods-archive-keyring.pgp \
+            --import \
+        && \
+    echo "deb [signed-by=/etc/apt/keyrings/renci-irods-archive-keyring.pgp arch=amd64] https://packages.irods.org/apt/ noble main" | \
+        tee /etc/apt/sources.list.d/renci-irods.list && \
+    wget -qO - https://core-dev.irods.org/irods-core-dev-signing-key.asc | \
+        gpg \
+            --no-options \
+            --no-default-keyring \
+            --no-auto-check-trustdb \
+            --homedir /dev/null \
+            --no-keyring \
+            --import-options import-export \
+            --output /etc/apt/keyrings/renci-irods-core-dev-archive-keyring.pgp \
+            --import \
+        && \
+    echo "deb [signed-by=/etc/apt/keyrings/renci-irods-core-dev-archive-keyring.pgp arch=amd64] https://core-dev.irods.org/apt/ noble main" | \
+        tee /etc/apt/sources.list.d/renci-irods-core-dev.list
```

Comment:
Ubuntu 24.04 did not include the core-dev repository before... but all the other projects touched in this commit did. Do we actually need core-dev in these? This is perhaps a question we can address some other time - it's fine for now.